aws/src/main/java/org/apache/iceberg/aws/s3/S3URI.java

```diff
@@ -70,19 +70,21 @@
 
     // Strip query and fragment if they exist
     String path = authoritySplit[1];
-    path = path.split(QUERY_DELIM)[0];
-    path = path.split(FRAGMENT_DELIM)[0];
+    path = path.split(QUERY_DELIM, -1)[0];
+    path = path.split(FRAGMENT_DELIM, -1)[0];
     this.key = path;
   }
 
   /**
+   * S3 bucket name
```

Comment:
The recommendation for these is to go with 
/**
 * Return s3 bucket
 */

So you still have a one liner, but it's the summary only

Reply:
Example from elsewhere int he codebase
```java
  /**
   * Returns the id of the field that the metrics within this class are associated with.
   */
  public int id() {
    return id;
  ```

Reply:
Thanks for the reference, updated!